homeassistant/components/nextdns/binary_sensor.py

```diff
@@ -54,21 +53,19 @@ class NextDnsBinarySensorEntityDescription(
 
 async def async_setup_entry(
     hass: HomeAssistant,
-    entry: ConfigEntry,
+    entry: NextDnsConfigEntry,
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Add NextDNS entities from a config_entry."""
-    coordinator: NextDnsConnectionUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
-        ATTR_CONNECTION
-    ]
+    coordinator = entry.runtime_data.connection
 
     async_add_entities(
         NextDnsBinarySensor(coordinator, description) for description in SENSORS
     )
 
 
 class NextDnsBinarySensor(
-    CoordinatorEntity[NextDnsConnectionUpdateCoordinator], BinarySensorEntity
+    CoordinatorEntity[NextDnsUpdateCoordinator[CoordinatorDataT]], BinarySensorEntity
 ):
```

Comment:
I don't really like the use of `CoordinatorDataT` here. It's basically resolved to any, so you don't get any additional validation from mypy that you passed the correct coordinator.

Instead, I'd suggest to use `NextDnsUpdateCoordinator[ConnectionStatus]` here. The same annotation should be applied in the `NextDnsData` class.

_Similarly for all other platforms, where possible._